org.osgi.test.junit4/src/test/java/org/osgi/test/junit4/context/BundleContextRuleTest.java

```diff
@@ -36,6 +39,37 @@
 
 public class BundleContextRuleTest {
 
+	@Test
+	public void testInstallBundle_B() throws Exception {
+		Bundle bundle = null;
+
+		try (WithContextRule it = new WithContextRule(getClass())) {
+			bundle = it.rule.installBundle("foo/tbfoo.jar", false);
+
+			assertThat(bundle).extracting(Bundle::getState)
+				.is(new Condition<Integer>(state -> (state & INSTALLED) == INSTALLED, "Installed"));
+		} finally {
+			assertThat(bundle).extracting(Bundle::getState)
```

Comment:
I am not sure I understand the idea of asserting in the finally block. If the assert in the try block throws an exception, the assert in the finally block can mask it by throwing its own exception, no? I get you are trying to test the try-with-resources uninstalls the bundles. But the bundle may not have properly installed and you want the assert failure. Asserting the try-with-resources properly cleaned up should just happen after the try-with-resources statement rather than in its finally block so that the finally assert cannot shadow a failed assert in the try-with-resources block.

Reply:
Is this a style complain? Does it make a significant difference?

Reply:
It is more than a style complaint. It is a complaint about shadowing assertion failures in the try block.